lib/matplotlib/_constrained_layout.py

```diff
@@ -140,6 +140,13 @@ def do_constrained_layout(fig, h_pad, w_pad,
                                     w_pad=w_pad, hspace=hspace, wspace=wspace)
                 else:
                     _api.warn_external(warn_collapsed)
+
+                if ((suptitle := fig._suptitle) is not None and
+                        suptitle.get_in_layout() and suptitle._autopos):
+                    x, _ = suptitle.get_position()
+                    suptitle.set_position(
+                        (x, layoutgrids[fig].get_inner_bbox().y1 + 0.02))
```

Comment:
Where does 0.02 come from? Does it work well with different font sizes?

Reply:
0.02 is the default space between the top of this title and the top of the figure without constrained layout, so I chose it to be consistent with that.

Reply:
I think all these look OK

![test_xx-large](https://github.com/user-attachments/assets/cbadb4c0-5f1f-452e-a1f5-970e42b48443)
![test_x-large](https://github.com/user-attachments/assets/add9c75e-ac01-40b1-86c8-119cd897d4e1)
![test_large](https://github.com/user-attachments/assets/1e15715b-0099-4850-8dbd-30cd51e0aed6)
![test_medium](https://github.com/user-attachments/assets/35ff9e4e-f9fe-4f4f-868b-b5490935bf7e)
![test_small](https://github.com/user-attachments/assets/6806b503-d7c9-4a37-81c2-ede692bc900a)

Reply:
Here's what happens with my current implementation if compressed layout is not moving axes in the vertical and I do not use `bbox_inches='tight'`.  I will change it to use _h_pad_.

![test_x-large(1)](https://github.com/user-attachments/assets/2cd4b235-e3c3-4438-ab9d-7f4f46caffc9)

Reply:
With the update, here is what I get with `figsize=(5, 5)` and no `bbox_inches='tight'`:
![test_x-large](https://github.com/user-attachments/assets/e824078b-123c-4f28-91bb-c1eed2078ffc)


Here is what I get with `figsize=(4, 10)` and saving with `bbox_inches='tight', pad_inches='layout'`:
![test_x-large(1)](https://github.com/user-attachments/assets/ff0978a3-018a-4fc3-b81c-f8f432722d09)